extension/app/inpage/ts/inpage.ts

```diff
@@ -194,52 +194,54 @@ class InterceptorMessageListener {
 			.catch(error => callback({ jsonrpc: '2.0', id: payload.id, error: { code: error.code, message: error.message, data: { ...error.data, stack: error.stack } } }, null))
 	}
 
-	private readonly WindowEthereumOn = async (kind: OnMessage, callback: AnyCallBack) => {
+	private readonly WindowEthereumOn = (kind: OnMessage, callback: AnyCallBack) => {
 		switch (kind) {
 			case 'accountsChanged':
 				this.onAccountsChangedCallBacks.add( callback as (accounts: readonly string[]) => void )
-				return
+				break
 			case 'message':
 				this.onMessageCallBacks.add(callback as (message: ProviderMessage) => void)
-				return
+				break
 			case 'connect':
 				this.onConnectCallBacks.add(callback as (connectInfo: ProviderConnectInfo) => void)
-				return
+				break
 			case 'close': //close is deprecated on eip-1193 by disconnect but its still used by dapps (MyEtherWallet)
 				this.onDisconnectCallBacks.add(callback as (error: ProviderRpcError) => void)
-				return
+				break
 			case 'disconnect':
 				this.onDisconnectCallBacks.add(callback as (error: ProviderRpcError) => void)
-				return
+				break
 			case 'chainChanged':
 				this.onChainChangedCallBacks.add(callback as (chainId: string) => void)
-				return
+				break
 			default:
```

Comment:
Since you are switching from `return` to `break` you should do an exhaustiveness check in `default`.  Alternatively, just `return window.ethereum` from each case.
```suggestion
			default: assertNever(kind)
```

Reply:
I tried to look from the docs what is the correct response, but metamask does not error on any strings